I3uckwheat/cards-of-carousal
# Feature/157 submit cards player

#### 0. Make sure your issue is linked:


closes #157 

#### 1. Purpose:


These changes ensure that the player's submitted card indexes are stored in the Host context and appropriate screens are displayed during the process.

#### 2. Implementation:


When a player selects their cards and hits "Submit", an action is dispatched with the selected card indexes as the payload.
Then the following happens:

1. A message is sent from the player to the host with the `player-submit` event and the selected card indexes as the payload
2. The player reducer updates the game state to display a waiting screen as the cards are submitted
3. The host dispatches the `PLAYER_SUBMIT` action which does two things:
   - Updates the player's submittedCards array (inside the Host context state) to be the array of indexes sent by the player
   - Sends an `update` message with the player as the recipient to let them know that their cards have been submitted
4. The player now sees a screen telling them to wait for other players

#### 3. Possible Issues:


Can't think of any right now

#### 4. How to Test:


1. Pull this branch
2. `PlayerScreenController.js`: Move `case: 'connected'` to be underneath `case: 'player-select'`
3. `PlayerContext.js`: Edit initial state to include dummy cards and increase selectCardCount
   - e.g. `cards: ['a', 'b', 'c']` and `selectCardCount: 2`
4. Run `npm run dev:backend` and then `npm run dev:frontend` in separate terminals
5. Open localhost and start a new game by selecting Host
6. Join the game by opening a new localhost tab and entering name/join code
7. You should now be able to select cards and hit Submit

#### 5. Cleanup
- [x] I've removed all un-necessary `console.log`s
